programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -1684,8 +1800,87 @@ static int get_mfa_codes_opts(int argc, char *argv[]) {
     return optind;
 }
 
-#if _WIN32
+static int set_log_level_opts(int argc, char *argv[]) {
+    static struct option opts[] = {
+            {"loglevel", required_argument, NULL, 'l'},
+    };
+    int c, option_index, errors = 0;
+    optind = 0;
+
+    tunnel_set_log_level *log_level_options = calloc(1, sizeof(tunnel_set_log_level));
+    cmd = calloc(1, sizeof(tunnel_comand));
```

Comment:
what happens to `cmd`? Unless I'm missing something it seems to be unused and leaked.

Reply:
Ahh. I see that it's a global. It's used and freed in `on_connect`.

Is there any way to be sure that on_connect is called exactly once when an option is changed? What prevents on_connect from trying to process the same `cmd` twice in row (e.g. a spurious connect to the pipe)?

Reply:
copied the data into the data parameter in the connect object